include/nbl/asset/utils/CPolygonGeometryManipulator.h

```diff
@@ -81,6 +81,81 @@ class NBL_API2 CPolygonGeometryManipulator
 				const_cast<IGeometryBase::SAABBStorage&>(geo->getAABBStorage()) = computeAABB(geo);
 		}
 
+		static inline core::smart_refctd_ptr<ICPUPolygonGeometry> createTriangleListIndexing(ICPUPolygonGeometry* geo, bool makeNewMesh = false)
```

Comment:
Take the input as const pointer, and use clone method to make the new mesh then replace the vertex and index data on that